cmds/basename/basename.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2012-2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Basename return name with leading path information removed.
+//
+// Synopsis:
+//     basename NAME [SUFFIX]
+//
+package main
+
+import (
+	"fmt"
+	"log"
+	"path"
+	"strings"
+
+	flag "github.com/spf13/pflag"
+)
+
+func usage() {
+	log.Fatal("Usage: basename NAME [SUFFIX]")
+}
+
+func main() {
+	flag.Parse()
+
+	args := flag.Args()
+	switch len(args) {
+	case 2:
+		_, fileName := path.Split(args[0])
```

Comment:
use filepath, not path package
use filepath.Base

Reply:
Here we see how broken the GNU commands are, btw: it's hard to cleanly do this
i=`basename /a/b/c`
since the command jams a newline into the result!